src/Components/CreateImageWizard/utilities/useValidation.tsx

```diff
@@ -122,6 +132,51 @@ export function useRegistrationValidation(): StepValidation {
     };
   }
 
+  if (registrationType === 'register-satellite') {
+    const errors = {};
+    if (caCertificate && (caCertificate === '' || !isValidPEM(caCertificate))) {
+      Object.assign(errors, {
+        certificate:
+          'Valid certificate must be present if you are registering Satellite.',
+      });
+    }
+    if (registrationCommand === '' || !registrationCommand) {
+      Object.assign(errors, {
+        command: 'No registration command for Satellite registration',
+      });
+    }
+    try {
+      const match = registrationCommand?.match(
+        /Bearer\s+([\w-]+\.[\w-]+\.[\w-]+)/
+      );
+      if (!match) {
+        Object.assign(errors, { command: 'Invalid token' });
+      } else {
+        const token = match[1];
+        const decoded = jwtDecode(token);
+        if (decoded.exp) {
+          const currentTime = Date.now() / 1000;
+          if (decoded.exp < currentTime) {
```

Comment:
Improvement idea, **not a blocker**: this could potentially have a buffer.
I mean the image build takes about 15 minutes, so token expiring in half an hour doesn't give you much room to use the image :stuck_out_tongue:
We might want to figure out a warning intead in a follow-up :)

Reply:
I like this idea, and I can do that as a followup. I still need to add the Satellite registration to the review step, add the command + certificate to the blueprint, so that it is saved. This could fit in too. :pencil2:  :notebook: